arbitrum-docs/launch-orbit-chain/how-tos/orbit-sdk-deploying-token-bridge.md

```diff
@@ -0,0 +1,180 @@
+
+1. **`createTokenBridgeEnoughCustomFeeTokenAllowance`**: This API is designed to verify that the deployer's address has enough allowance to pay for the fees associated with the bridge token deployment.
+2. **`createTokenBridgePrepareCustomFeeTokenApprovalTransactionRequest`**: This API assists in generating the raw transaction required to approve the native token for the `TokenBridgeCreator` contract.
+
+The following example demonstrates how to leverage these APIs effectively to check for and, if necessary, grant approval to the `TokenBridgeCreator` contract:
+
+```js
+const allowanceParams = {
+  nativeToken,
+  owner: rollupOwner.address,
+  publicClient: parentChainPublicClient,
+};
+if (!(await createTokenBridgeEnoughCustomFeeTokenAllowance(allowanceParams))) {
+  const approvalTxRequest =
+    await createTokenBridgePrepareCustomFeeTokenApprovalTransactionRequest(allowanceParams);
+}
+```
+
+In this scenario, `allowanceParams` includes the native token details, the roll-up owner's address, and the public client for the parent chain. Initially, the `createTokenBridgeEnoughCustomFeeTokenAllowance` API checks if the deployer's allowance is sufficient. If the allowance is inadequate, the `createTokenBridgePrepareCustomFeeTokenApprovalTransactionRequest` API is called to create the necessary approval transaction.
+
+It is important to note that after generating the raw transaction, the deployer must sign and broadcast it to the network to finalize the approval process.
+
+### 2. Token Bridge Contract Deployment{#step-2}
+
+The deployment of token bridge contracts constitutes the foundational step in establishing a bridge between the parent and child chains. This process mirrors the deployment methodology used for orbit chain contracts, where a primary contract, named `RollupCreator`, facilitates the deployment of core contracts. In the context of token bridge contracts, the `TokenBridgeCreator` contract assumes a similar pivotal role by orchestrating the deployment across both the parent and child chains. Feel free to take a look at the [`TokenBridgeCreator` Solidity code](https://github.com/OffchainLabs/token-bridge-contracts/blob/b3894ecc8b6185b2d505c71c9a7851725f53df15/contracts/tokenbridge/ethereum/L1AtomicTokenBridgeCreator.sol#L4).
+
+`TokenBridgeCreator` can deploy the token bridge contracts on both the parent and child chains in a single transaction. A common query is how it manages to directly deploy contracts on the child chain from the parent chain. The solution lies in using the Retryable Tickets protocol, which facilitates the transmission of the deployment message between the two chains. This message, once transmitted, triggers the contract deployment by the Retryable Tickets mechanism. For an in-depth understanding, please refer to this [explanation of the Retryable Ticket system](https://docs.arbitrum.io/arbos/l1-to-l2-messaging#retryable-tickets).
```

Comment:
We should use relative URLs, instead of specifying docs.arbitrum.io

Reply:
Merged manually.